modules/n1ql/pages/n1ql-language-reference/createfunction.adoc

```diff
@@ -198,36 +244,46 @@ image::n1ql-language-reference/create-function-external.png["Syntax diagram: ref
 
 // TODO: Automatic links in EBNF.
 
-function:: (Required) Refer to <<external-name>> below.
+[horizontal.compact]
+function:: <<external-name>> icon:caret-down[]
+params:: <<external-parameter>> icon:caret-down[]
+obj:: <<external-object>> icon:caret-down[]
+library:: <<external-library>> icon:caret-down[]
+javascript:: <<javascript>> icon:caret-down[]
 
-params:: (Optional) Refer to <<external-parameter>> below.
-
-obj:: (Required) Refer to <<external-object>> below.
-
-library:: (Required) Refer to <<external-library>> below.
+include::createfunction.adoc[tag=replace]
 
 include::createfunction.adoc[tag=arguments]
 
 [[external-object]]
 ==== External Object
 
+[Optional] Use this parameter where the function code is stored in an external library.
+
 The name of the JavaScript function that you want to use for the user-defined function.
 This parameter is a string and must be wrapped in quotes.
-// FIXME:
-External functions in {sqlpp} _only_ support plain JavaScript, without any of the xref:eventing:eventing-language-constructs.adoc#added-lang-features[added language features] supported by functions in the Eventing Service.
 
 [[external-library]]
 ==== External Library
 
-The name of the JavaScript library that contains the function you want to use.
+[Optional] Use this parameter where the function code is stored in an external library.
+
+The name of the JavaScript library that contains the JavaScript function you want to use.
 This parameter is a string and must be wrapped in quotes.
-You must create the JavaScript library and the JavaScript function using the {sqlpp} Functions REST API.
-For details, refer to xref:n1ql-rest-api/functions.adoc[Functions REST API].
 
 The name of a scoped external library must include the bucket name, the scope name, and the library name, separated by slashes.
 For example, to refer to a scoped library called `my-library` located in the `inventory` scope within the `travel-sample` bucket, you would specify the library name as `travel-sample/inventory/my-library`.
 
-include::createfunction.adoc[tag=replace]
+[[javascript]]
+==== Function Body
+
+[Optional] Use this parameter to create a {sqlpp} managed user-defined function.
+
+The external JavaScript function code.
+This must contain a function with the same name and the same number of parameters as the {sqlpp} user-defined function.
+This parameter is a string and must be wrapped in quotes.
+
+The JavaScript code can contain multiple function definitions, but these functions can only be referenced within this {sqlpp} user-defined function, and cannot be shared.
```

Comment:
Very minor change request - 

"But these functions can only be referenced within the Javascript code for this user-defined function and cannot be shared"

To

"The JavaScript code can contain multiple function definitions, but these functions can only be referenced within the Javascript code for this user-defined function, and cannot be shared."


[ The reason I ask is because SQL++ user defined function implies the corresponding SQL++ function and not the actual Javascript code tied to it. ]

Reply:
Done!